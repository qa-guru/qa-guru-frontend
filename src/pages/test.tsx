import { FC } from "react";
import { useParams } from "react-router-dom";

import TestContainer from "features/test/containers/test-container";

const TestPage: FC = () => {
  const { testId, trainingId, lectureId } = useParams<{
    testId: string;
    trainingId: string;
    lectureId: string;
  }>();

  // Проверяем, что все параметры присутствуют
  if (!testId || !trainingId || !lectureId) {
    return <div>Ошибка: отсутствуют необходимые параметры</div>;
  }

  return (
    <TestContainer
      testId={testId}
      trainingId={trainingId}
      lectureId={lectureId}
    />
  );
};

export default TestPage;
