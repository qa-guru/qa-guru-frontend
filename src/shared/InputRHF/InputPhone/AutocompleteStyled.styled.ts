import { styled } from "@mui/system";
import { Autocomplete } from "@mui/material";

export const AutocompleteStyled = styled(Autocomplete)(({ theme }) => ({
  [theme.breakpoints.down("md")]: {
    "& .MuiOutlinedInput-root": {
      padding: "5px",
    },
  },
}));
