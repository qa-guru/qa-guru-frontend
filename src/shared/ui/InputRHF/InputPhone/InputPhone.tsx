import { Controller } from "react-hook-form";
import { IFormInputProps } from "../Input.types";
import React from "react";
import "react-phone-number-input/style.css";
import PhoneInput from "react-phone-number-input";

const InputPhone: React.FC<IFormInputProps> = ({ control, name }) => {
  return (
    <Controller
      name={name}
      control={control}
      render={({ field: { onChange, value } }) => (
        <PhoneInput
          international
          countryCallingCodeEditable={false}
          defaultCountry="RU"
          value={value}
          onChange={onChange}
        />
      )}
    />
  );
};

export default InputPhone;
