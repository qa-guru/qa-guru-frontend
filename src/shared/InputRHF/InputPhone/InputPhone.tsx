import { Controller } from "react-hook-form";
import { IFormInputProps } from "../Input.types";
import React from "react";
import { Autocomplete, Box } from "@mui/material";
import { countries } from "./InputPhone.countries";
import { styled } from "@mui/system";
import { TextFieldStyled } from "../InputTextField/InputTextField.styled";

const AutocompleteStyled = styled(Autocomplete)(({ theme }) => ({
  [theme.breakpoints.down("md")]: {
    "& .MuiOutlinedInput-root": {
      padding: "5px",
    },
  },
}));

const InputPhone: React.FC<IFormInputProps> = ({
  control,
  name,
  label,
  placeholder,
}) => {
  return (
    <Controller
      name={name}
      control={control}
      render={({ field: { value, onChange } }) => (
        <AutocompleteStyled
          options={countries}
          freeSolo
          getOptionLabel={(option: any) => `(${option.code})+${option.phone}`}
          renderOption={(props, option: any) => (
            <Box
              component="li"
              sx={{ "& > img": { mr: 2, flexShrink: 0 } }}
              {...props}
            >
              <img
                loading="lazy"
                width="20"
                src={`https://flagcdn.com/w20/${option.code.toLowerCase()}.png`}
                srcSet={`https://flagcdn.com/w40/${option.code.toLowerCase()}.png 2x`}
                alt="flag"
              />
              {option.label} ({option.code}) +{option.phone}
            </Box>
          )}
          renderInput={(params) => (
            <TextFieldStyled
              value={value}
              onChange={onChange}
              {...params}
              inputProps={{
                ...params.inputProps,
              }}
              label={label}
              placeholder={placeholder}
            />
          )}
        />
      )}
    />
  );
};

export default InputPhone;
