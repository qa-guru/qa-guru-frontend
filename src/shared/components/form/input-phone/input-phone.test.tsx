import { MemoryRouter } from "react-router-dom";
import { FieldErrors, FormProvider, useForm } from "react-hook-form";
import { render } from "test/utils-test";

import InputPhone from "./input-phone";

describe("InputPhone", () => {
  it("renders correctly", () => {
    function TestComponent() {
      const methods = useForm();
      const errors: FieldErrors = {};
      return (
        <MemoryRouter>
          <FormProvider {...methods}>
            <InputPhone
              control={methods.control}
              name="testDate"
              label="testDate"
              placeholder="testDate"
              errors={errors}
            />
          </FormProvider>
        </MemoryRouter>
      );
    }

    const { asFragment } = render(<TestComponent />);

    expect(asFragment()).toMatchSnapshot();
  });
});
