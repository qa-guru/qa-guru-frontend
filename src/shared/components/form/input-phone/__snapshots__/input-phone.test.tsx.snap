// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`InputPhone > renders correctly 1`] = `
<DocumentFragment>
  <div
    class="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
  >
    <div
      class="MuiAutocomplete-root css-1h51icj-MuiAutocomplete-root"
    >
      <div
        class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-1r4resw-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="false"
          for=":r0:"
          id=":r0:-label"
        >
          testDate
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiAutocomplete-inputRoot css-tc0ri2-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <input
            aria-autocomplete="list"
            aria-expanded="false"
            aria-invalid="false"
            autocapitalize="none"
            autocomplete="off"
            class="MuiInputBase-input MuiOutlinedInput-input MuiAutocomplete-input MuiAutocomplete-inputFocused css-1lkr5rz-MuiInputBase-input-MuiOutlinedInput-input"
            id=":r0:"
            placeholder="testDate"
            role="combobox"
            spellcheck="false"
            type="text"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-yjsfm1"
            >
              <span>
                testDate
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
